spec/01-unit/04-utils_spec.lua

```diff
@@ -98,29 +98,42 @@ describe("Utils", function()
       it("should validate any scheme with X-Forwarded_Proto as HTTPS", function()
         headers["x-forwarded-proto"] = "hTTPs"  -- check mixed casing for case insensitiveness
         ngx.var.scheme = "hTTps"
-        assert.is.truthy(utils.check_https(true))
+        assert.is.truthy(utils.check_https(true, true))
         ngx.var.scheme = "hTTp"
-        assert.is.truthy(utils.check_https(true))
+        assert.is.truthy(utils.check_https(true, true))
         ngx.var.scheme = "something completely different"
-        assert.is.truthy(utils.check_https(true))
+        assert.is.truthy(utils.check_https(true, true))
       end)
 
       it("should validate only https scheme with X-Forwarded_Proto as non-HTTPS", function()
         headers["x-forwarded-proto"] = "hTTP"
         ngx.var.scheme = "hTTps"
-        assert.is.truthy(utils.check_https(true))
+        assert.is.truthy(utils.check_https(true, true))
         ngx.var.scheme = "hTTp"
-        assert.is.falsy(utils.check_https(true))
+        assert.is.falsy(utils.check_https(true, true))
         ngx.var.scheme = "something completely different"
-        assert.is.falsy(utils.check_https(true))
+        assert.is.falsy(utils.check_https(true, true))
       end)
 
       it("should return an error with multiple X-Forwarded_Proto headers", function()
         headers["x-forwarded-proto"] = { "hTTP", "https" }
         ngx.var.scheme = "hTTps"
-        assert.is.truthy(utils.check_https(true))
+        assert.is.truthy(utils.check_https(true, true))
         ngx.var.scheme = "hTTp"
-        assert.are.same({ nil, "Only one X-Forwarded-Proto header allowed" }, { utils.check_https(true) })
+        assert.are.same({ nil, "Only one X-Forwarded-Proto header allowed" },
+                        { utils.check_https(true, true) })
+      end)
+
+      it("should not use X-Forwarded-Proto when the client is untrusted", function()
+        headers["x-forwarded-proto"] = "https"
+        ngx.var.scheme = "http"
+        assert.is_false(utils.check_https(false, false))
+        assert.is_false(utils.check_https(false, true))
+
+        headers["x-forwarded-proto"] = "https"
+        ngx.var.scheme = "https"
+        assert.is_true(utils.check_https(false, false))
+        assert.is_true(utils.check_https(false, true))
```

Comment:
It'd be nice to also have:
```
should use X-Forwarded-Proto when the client is trusted
```

With the proper capitalization of `https` (unlike the tests ensuring the case-insensitiveness). Doing so ensures good atomicity of the features being tested, as case-insensitiveness might be broken, but not the `== "https"` check altogether.